client/src/main/java/org/apache/falcon/ExtensionHandler.java

```diff
@@ -186,4 +192,27 @@ private static String createStagePath(String extensionName, String jobName) {
         urls.add(fileURL);
         return urls;
     }
+
+
+    public static String getExtensionLocation(String extensionName, JSONObject extensionDetailJson) {
```

Comment:
This method has copy-paste issues. Replace all references to type with location.